src/CalcManager/Ratpack/logic.cpp

```diff
@@ -172,47 +172,93 @@ void boolnum( PNUMBER *pa, PNUMBER b, int func )
         case FUNC_XOR:
             *pchc++ = da ^ db;
             break;
-            }
         }
+    }
     c->sign = a->sign;
-    while ( c->cdigit > 1 && *(--pchc) == 0 )
-        {
+    while (c->cdigit > 1 && *(--pchc) == 0)
+    {
         c->cdigit--;
-        }
-    destroynum( *pa );
-    *pa=c;
+    }
+    destroynum(*pa);
+    *pa = c;
 }
 
 //-----------------------------------------------------------------------------
 //
-//    FUNCTION: modrat
+//    FUNCTION: remrat
 //
 //    ARGUMENTS: pointer to a rational a second rational.
 //
 //    RETURN: None, changes pointer.
 //
-//    DESCRIPTION: Does the rational equivalent of frac(*pa);
+//    DESCRIPTION: Calculate the remainder of *pa / b,
+//                 equivalent of 'pa % b' in C/C++ and produces a result
+//                 that is either zero or has the same sign as the dividend.
 //
 //-----------------------------------------------------------------------------
 
-void modrat( PRAT *pa, PRAT b )
+void remrat(PRAT *pa, PRAT b)
```

Comment:
Why does this need to be a `PRAT*`?  Can't it be a `PRAT`?

Reply:
because the result of pa % b is stored in `pa`

Reply:
While PRAT is just a typedef'd RAT* and can therefore be modified as a reference type, the RatPack uses PRAT as though it were a value-type, passing PRAT* to functions that will modify the value. For consistency, I think sticking with that precedent makes sense.